psyneulink/core/components/component.py

```diff
@@ -3223,8 +3227,12 @@ def fill_recursively(arr, value, indices=()):
         except AttributeError:
             pass
 
-        fill_recursively(var, param)
-        return var
+        param_arr = np.asarray(param)
+        if param_arr.dtype == int or param_arr.dtype == float:
```

Comment:
can this use `np.isscalar` ?

Reply:
Not quite, np.isscalar returns False for 0-dim numeric arrays (and longer), though in many cases they can be reshaped to match variable.

Reply:
I meant `np.isscalar(param_arr.dtype)`, that should cover both `int` and `float`

Reply:
Oh, I see. The main point is to check for a fully regular, numeric array. I suppose `np.unsignedinteger` would be a valid dtype though I don't think we use it anywhere. In this case, maybe it makes sense to just rewrite as `if param_arr.dtype != object`, because any other invalid types for param/noise should be instead caught at initial assignment and would fail later regardless. I have already used that for a similar check elsewhere